cogs/stockmarket/market.py

```diff
@@ -0,0 +1,568 @@
+
+            condition = MessagePredicate.positive(ctx, None, ctx.author)
+
+            try:
+                message = await ctx.bot.wait_for("message", check=condition, timeout=35.0)
+            except asyncio.TimeoutError:
+                new_embed = discord.Embed(color=0xFF0000, title=const.BROKER_NAME, description=const.TRANSACTION_CANCELLED_STR)
+                await sent_message.edit(embed=new_embed)
+                return
+            amount_to_buy = int(message.content)
+
+            if 0 < amount_to_buy <= maximum_purchasable:
+                break
+
+            if amount_to_buy == 0:
+                new_embed = discord.Embed(color=0xFF0000,
+                                          title=const.BROKER_NAME,
+                                          description=const.TRANSACTION_CANCELLED_STR)
+                await sent_message.edit(embed=new_embed)
+                return
+
+        new_embed = discord.Embed(color=0x00FF00,
+                                  title=const.BROKER_NAME,
+                                  description='Shares Purchased!')
+        await sent_message.edit(embed=new_embed)
+        await bank.withdraw_credits(ctx.author, amount_to_buy * price)
+
+        # add stock to portfolio
+        # Could add more details to each holdings, like average price etc.
+        async with self.config.member(ctx.author).holdings() as holdings:
+            if ticker not in holdings:
+                holdings.append({'ticker': ticker, 'amount': amount_to_buy, 'initial DOP': time.time()})
+            else:
+                for holding in holdings:
+                    if holding['ticker'] == ticker:
+                        holding['amount'] += amount_to_buy
+                        break
+
+        async with self.config.member(ctx.author).transactions() as transactions:
+            transactions.append({
+                'type': 'buy',
+                'ticker': ticker,
+                'date': datetime.utcfromtimestamp((int)(time.time())).strftime('%Y-%m-%d %H:%M:%S'),
+                'shares_bought': amount_to_buy,
+                'bought_price': price,
+                'amount_recieved': amount_to_buy * -price
+            })
+
+    @market.command(name="holdings")
+    async def list_holdings(self, ctx, user_id: discord.Member=None):
+        """Lists all of your stocks and everything else that is in your portfolio"""
+
+        async with self.config.member(ctx.author).holdings() as holdings:
+            if not holdings:
+                await ctx.send(f'{ctx.author.mention} you do not own any shares!')
+                return
+
+            # port_value = 0
+            msg = 'You own the following stocks:\n ```css\n'
+
+            print(holdings)
+
+            for holding in holdings:
+                ticker = holding['ticker']
+                amount = holding['amount']
+                msg += f'[{amount}] shares of [{ticker}]\n'

```

Comment:
Consider constructing this first and then using `redbot.core.utils.chat_formatting.box` for backticks